src/controllers/emotes/EmoteHolder.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "common/Aliases.hpp"
+
+#include <pajlada/signals/signal.hpp>
+#include <pajlada/signals/signalholder.hpp>
+
+#include <memory>
+#include <span>
+#include <vector>
+
+namespace chatterino {
+
+struct Emote;
+using EmotePtr = std::shared_ptr<const Emote>;
+class EmoteProvider;
+class EmoteMap;
+using EmoteMapPtr = std::shared_ptr<const EmoteMap>;
+class EmoteController;
+class EmoteChannel;
+
+class Channel;
+using ChannelPtr = std::shared_ptr<Channel>;
+
+/// An entry for a single provider
+struct EmoteHolderItem {
```

Comment:
I don't like the term "item" or "entry", at least not when it's this generic.
How about something like "provider handle"?